src/Editor/GUI/Editors/EntityEditor.cpp

```diff
@@ -355,65 +356,127 @@ void EntityEditor::RigidBodyEditor(Component::RigidBody* rigidBody) {
+
+                int typeId = script->instance->GetPropertyTypeId(n);
+                void *varPointer = script->instance->GetAddressOfProperty(n);
+                if (typeId == asTYPEID_INT32)
+                {
+                    ImGui::InputInt(script->instance->GetPropertyName(n), (int*)script->propertyMap[script->instance->GetPropertyName(n)][typeId], 0.0f);
+                }
+                else if (typeId == asTYPEID_FLOAT)
+                {
+                    ImGui::DraggableFloat(script->instance->GetPropertyName(n), *(float*)script->propertyMap[script->instance->GetPropertyName(n)][typeId], 0.0f);
+                }
+                else if (typeId & asTYPEID_SCRIPTOBJECT)
                 {
-                    script->refList.push_back(entity);
+                    asIScriptObject *obj = (asIScriptObject*)varPointer;
                 }
+                else if (typeId == script->instance->GetEngine()->GetTypeIdByDecl("string"))
+                {
+                    std::string *str = (std::string*)script->propertyMap[script->instance->GetPropertyName(n)][typeId];
+                    if (str) {
+
+                        if (str->size() <= 128) {
+
+                            std::copy(str->begin(), str->end(), stringPropertyBuffer);
+                            stringPropertyBuffer[str->size()] = '\0';
+
+                            ImGui::InputText(script->instance->GetPropertyName(n), stringPropertyBuffer, 128);
+
+                            *str = stringPropertyBuffer;
+
+                        }
+                        else
+                            ImGui::Text("%s = <TOO BIG>\n", script->instance->GetPropertyName(n));
+
+                    }
+                    else
+                        ImGui::Text("%s = <null>\n", script->instance->GetPropertyName(n));
+                }
+
+            }
+
+            if (ImGui::Button("Reset properties")) {
+
+                Managers().scriptManager->ClearPropertyMap(script);
+                Managers().scriptManager->FillPropertyMap(script);
+
             }
 
-            ImGui::EndPopup();
         }
-    }
-    else
-        ImGui::Text("No script loaded");
+        ImGui::Separator();
 
+		ImGui::Text("Entity References");
+		// Display current entity references
+		for (size_t i = 0; i != script->refList.size(); ++i) {
+			ImGui::Text(script->refList[i]->name.c_str());
+			ImGui::SameLine(ImGui::GetWindowWidth() - 30);
+			if(ImGui::SmallButton(("x###remove" + std::to_string(i)).c_str())){
+				script->refList.erase(script->refList.begin() + i);
+				break;
+			}
+		}
+
+		// Choosing other entity references
+		if (ImGui::Button("Add entity reference")) {
+			ImGui::OpenPopup("Add entity reference");
+		}
+
+		if (ImGui::BeginPopup("Add entity reference"))
+		{
+			ImGui::Text("Entities");
+			ImGui::Separator();
+			for (Entity* entity : Hymn().world.GetEntities()) //Change into a prettier tree structure or something, later.
+			{
```

Comment:
No braces with single statement body. The comment above could be placed in a doxygen todo comment so that it's added to a todo-list automatically.